diff --git a/pkgs/apps/freqmine/src/data.cpp b/pkgs/apps/freqmine/src/data.cpp
index aa1454b..7c82fc3 100644
--- a/pkgs/apps/freqmine/src/data.cpp
+++ b/pkgs/apps/freqmine/src/data.cpp
@@ -9,9 +9,9 @@ Contents : data set management
 Data::Data(char *filename)
 {
 #ifndef BINARY
-  in = fopen(filename,"r+t");
+  in = fopen(filename,"rt");
 #else
-  in = fopen(filename, "r+b");
+  in = fopen(filename, "rb");
 #endif
 }
 
