diff --git a/config/gcc.bldconf b/config/gcc.bldconf
index 890efbb..3481aeb 100644
--- a/config/gcc.bldconf
+++ b/config/gcc.bldconf
@@ -39,9 +39,9 @@ case "${OSTYPE}" in
   # Linux + default case
 
   # CC_HOME is installation root of the C compiler
-  export CC_HOME="/n/fs/parsec/local/gcc-4.4.0-static"
+  export CC_HOME="/usr"
   # BINUTIL_HOME is installation root of the GNU binutils
-  export BINUTIL_HOME="/usr/local"
+  export BINUTIL_HOME="/usr"
   # GNUTOOL_HOME is installation root of the GNU tools
   export GNUTOOL_HOME="/usr"
   # BINARY_PREFIX is the string which is used as prefix for the GNU binaries
